### PR TITLE
feat: set protocol according to app.options.https

### PR DESCRIPTION
**Pull request by cyjake**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

egg-view-assets itself

##### Description of change

If user customizes config `assets.url`, retain the other parts of `assets.url` by only replacing `port`.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -19,7 +20,9 @@ module.exports = app => {
         throw new Error('check autoPort fail');
       }
     }
-    assetsConfig.url = 'http://127.0.0.1:' + port;
+    const url = new URL(assetsConfig.url || 'http://127.0.0.1');
```
Comment (popomore) [inline app.js]: 如果只是判断协议的话是否直接读取 app.options.https，然后自动切换

Comment (cyjake) [reply app.js]: 可以呀，我改改

File: test/assets.test.js [hunk 2]
```diff
@@ -243,6 +244,34 @@ describe('test/assets.test.js', () => {
...
+    after(() => app.close());
+
+    it('should GET /', () => {
+      return urllib.request('https://127.0.0.1:8443', {
+        dataType: 'text',
+        rejectUnauthorized: false,
+      }).then(response => {
+        assert(response.status === 200);
+        assert(response.data.includes('https:\/\/127.0.0.1:8000\/index.css'));
+      });
```
[the remark this answers is not in the document]
Comment (popomore) [reply test/assets.test.js]: 因为是 supertest 的某些黑科技

**Comment by popomore** [comment]
+ egg-view-assets@1.6.0

**Pull request merged**